lib/decorators/inject.ts

```diff
@@ -75,5 +76,27 @@ export function Inject( ...injects: any[] ){
 		else {
 			bundleStore.set('$inject', dependencies, t);
 		}
+
+		componentHooks.beforeCtrlInvoke(injectParentComponents);
 	}
 }
+
+// Checks every injection if it is known to the $injector
+// It will then check the unknown injects if there is any parent component that matches
+// the name. If a matching component was found it will inject that component as a
+// local.
+function injectParentComponents(caller: any, injects: string[], controller: any, ddo: any, $injector: any, locals: any) {
+	injects.forEach((inject) => {
+		if (!$injector.has(inject)) {
```

Comment:
NVM!! I was gonna say "Instead of `!$injector.has(inject)` let's do `providerStore.get('type', controller) === 'component'`. This way only classes that have been decorated with @Component go into this if block." 

But then I realized that this is the inject of **strings**... not the classes, so we can't make this check. I think what you have is good after realizing this.